pkg/cli/mirror/additional.go

```diff
@@ -36,7 +36,7 @@ func (o *AdditionalOptions) Plan(ctx context.Context, imageList []v1alpha2.Image
 		if err != nil {
 			return mmappings, fmt.Errorf("error parsing source image %s: %v", img.Name, err)
 		}
-		srcRef.Ref = srcRef.Ref.DockerClientDefaults()
+		srcRef.Ref = image.SetDockerClientDefaults(srcRef.Ref)
```

Comment:
Based on https://issues.redhat.com/browse/OCPBUGS-2633 and my understanding this only applies when working with files, can you ensure this does not bleed into other areas?

Reply:
It kinda does affect it, though less seriously. This is the output of a MirrorToMirror with a similar imageSetConfig as above (4 imgs instead of 2):
```
╰─ (main)-$ ./bin/oc-mirror -c ocpbugs-2633/isc.yaml docker://localhost:5000/ocpbugs2633 --dest-skip-tls --dest-use-http
Checking push permissions for localhost:5000
Found: oc-mirror-workspace/src/publish
Found: oc-mirror-workspace/src/v2
Found: oc-mirror-workspace/src/charts
Found: oc-mirror-workspace/src/release-signatures
backend is not configured in ocpbugs-2633/isc.yaml, using stateless mode
backend is not configured in ocpbugs-2633/isc.yaml, using stateless mode
No metadata detected, creating new workspace
localhost:5000/
  ocpbugs2633/okd/scos-content
    blobs:
      quay.io/okd/scos-content sha256:d8190195889efb5333eeec18af9b6c82313edd4db62989bd3a357caca4f13f0e 1.404KiB
      quay.io/okd/scos-content sha256:9578bde7a5452e36c2da7dca455705194e76d896b3d3371358fc3fc027e203d4 1.42KiB
      quay.io/okd/scos-content sha256:24e7a0609d80240d989307508f887c4e9d96787c9b0b5378b5b45be0125413f1 16.11KiB
      quay.io/okd/scos-content sha256:6f70be2ee17ebb8c71b862a5f364a8c9cd39901fd2c039dc226152bca81e4bfd 23.09KiB
      quay.io/okd/scos-content sha256:6277cd34dc835d34ece600a35ff14bd9351a2fc27bce21335c633d6c314496cb 23.26KiB
      quay.io/okd/scos-content sha256:97a6159bc677aa2abbc3d0222cdbb9be39615af67c88b46f1d9aa0150a83f47a 23.64KiB
      quay.io/okd/scos-content sha256:716e739ca7cb8dbc09187be2bdeed93262653996e0b13f5936e4e909ef99a4b4 463.2KiB
      quay.io/okd/scos-content sha256:50c5caf281eb1792409d447284cca3df93b1f0c5f56c8ff34c0fa5708db44f0d 465.1KiB
      quay.io/okd/scos-content sha256:5ca70b43221ac56d26d9b3a024d2535fd945505b3e6b3135360eaedb7ca47d83 469.7KiB
      quay.io/okd/scos-content sha256:258378b4147c91df6dacf46553dc16aafbff18c67297fb9dc8f04103fef4b170 476.1KiB
      quay.io/okd/scos-content sha256:2e7c4173d931d7cc039747f0790ed11fba42f964d8e7d0b05b7808e782ed0564 11.79MiB
      quay.io/okd/scos-content sha256:622ceb2281df8e15e9ffdfb635b2eb3e00360b7aee3e47c67e6fdc747471133a 11.79MiB
      quay.io/okd/scos-content sha256:dbd260b8a6ecf4f527d431cb1525a427f9674e00050e8251ce9f9d138e39cb67 11.83MiB
      quay.io/okd/scos-content sha256:67da1a739c8f7bda5695f7341290728b043fde552169340fb8ed6837429556a6 12.12MiB
      quay.io/okd/scos-content sha256:dfaf6e5198c10186790f5f02d3d4e18941704c0f29b0dda3e7a1df12563a5c5a 21.71MiB
      quay.io/okd/scos-content sha256:0e8721c30137905641e58cf7943f058a1eb679814c8ad3541bd85892bf7667c9 24.62MiB
      quay.io/okd/scos-content sha256:067f126180dff2d087953bfe776a38a1b5923fcffd5bbe13074938df2ef67a3d 25.44MiB
      quay.io/okd/scos-content sha256:66667e1e64ec0550da478c8fffebca5ef763064d3ec8e0bbd530cd4f9ee0be91 26.09MiB
      quay.io/okd/scos-content sha256:4534f7f2788cb7ed1abd839410c90e8fb95d2145c5add2d0646021c546a75786 30.87MiB
      quay.io/okd/scos-content sha256:32998f14f53a785b31ab5c3d9daf614f288e2c0d018099f83b5c2e6598078de1 40.49MiB
      quay.io/okd/scos-content sha256:69788635f90cb66139f7d62ff431c784b4fd7b98632bb18e7cc10bebab55205e 70.46MiB
      quay.io/okd/scos-content sha256:97da74cc6d8fa5d1634eb1760fd1da5c6048619c264c23e62d75f3bf6b8ef5c4 75.84MiB
    manifests:
      sha256:4e2ec95cb15bf41985dd4e7701c5f8aa82cae0d074552e68353f6330fb26ffd7 -> latest
      sha256:df80aa07467d1c6f59a39f3c00e00e130a6b25308b1419264565ca7cd8a76407 -> latest
      sha256:fa2453bdf4a7e4ce61767e9d0ec0d36d1a08819afae6c884adef189799579435 -> latest
      sha256:fc37fb091804ce32411d04559a4b0ba63139bd12b51f7d87dc2e8fa9ff9d3ef7 -> latest
  stats: shared=0 unique=22 size=365MiB ratio=1.00

phase 0:
  localhost:5000 ocpbugs2633/okd/scos-content blobs=22 mounts=0 manifests=4 shared=0

info: Planning completed in 1.07s
```
As you can see, the `latest` label will just get "moved" 4 times.